src/Core/SonarArtefactKeys.cs

```diff
@@ -0,0 +1,73 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+
+namespace SonarLint.VisualStudio.Core
+{
+    // Concepts:
+    // The Sonar plugin model defines various types of artefact, all of which have
+    // unique identifiers:
+    //
+    // * plugin : a container for different types of extension
+    //
+    // * language : A language is defined in a single plugin but can be consumed by multiple
+    //              e.g. the CSharp language is defined in the SonarC# plugin, with additional
+    //              rules for the language being defined in the SonarC# Security plugin.
+    //
+    // * repository: a container for rule definitions. A repository provides rules for a single language.
+    //
+    // * rule definition: part of a single repository. The full rule id is given by "[repo id]:[rule id]"
+
+    /// <summary>
+    /// Language keys for languages supported by SonarQube/Cloud plugins
+    /// </summary>
+    /// <remarks>A full list of languages keys can be obtained by calling https://sonarcloud.io/api/languages/list
+    /// </remarks>
+    public static class SonarLanguageKeys
+    {
+        // TODO: the SonarQube.Client assembly has a class that defines the SonarQube language keys:
```

Comment:
This type is unchanged (except for the addition of the TODO comment - perhaps we should move the definition of all of the server constants to the SonarQube.Client assembly).

The other types are new.